contracts/upgradeables/chips/Game.sol

```diff
@@ -0,0 +1,170 @@
+        _setRoleAdmin(DEFAULT_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
+        _setRoleAdmin(GAME_SERVER_ROLE, DEFAULT_ADMIN_ROLE);
+
+        chips = IChips(_chips);
+        treasury = _treasury;
+        defaultPlayCost = _defaultPlayCost;
+        if (_isPaused) _pause();
+    }
+
+    function buyPlays(
+        address _player,
+        uint256 _gameNumber,
+        uint256 _numPlays
+    ) external onlyRole(GAME_SERVER_ROLE) whenNotPaused nonReentrant {
+        uint256 gamePlayCost = _getPlayCost(_gameNumber);
+        uint256 buyinAmount = _numPlays * gamePlayCost;
+        
+        uint256 initialBalance = chips.balanceOf(_player);
+        if (initialBalance < buyinAmount) {
+            revert InsufficientChipBalance(_player, buyinAmount);
+        }
+        
+        chips.retrieveBuyIn(_player, buyinAmount);
+        totalGameValue[_gameNumber] += buyinAmount;
+        currentGameValue[_gameNumber] += buyinAmount;
+
+        playBalance[_gameNumber][_player] += _numPlays;
+        emit PlaysBought(_player, _gameNumber, _numPlays);
+    }
+        
+    function payout(
+        uint256 _gameNumber,
+        address[] calldata _players,
+        uint256[] calldata _prizeValues,
+        uint256 _rake
+    ) external onlyRole(GAME_SERVER_ROLE) whenNotPaused nonReentrant {
+        require(
+            _players.length == _prizeValues.length,
+            "Game.payout: incongruent number of players and prizes"
+        );
+        require(treasury != address(0), "Treasury address not set");
+
+        address[] memory receipients = new address[](_players.length + 1);
+        for (uint256 i = 0; i < _players.length; i++) {
+            receipients[i] = _players[i];
+        }
+        receipients[_players.length] = treasury; // add the new value at the end
+
+        uint256[] memory payoutValues = new uint256[](_prizeValues.length + 1);
+        for (uint256 i = 0; i < _prizeValues.length; i++) {
+            payoutValues[i] = _prizeValues[i];
+        }
+
+        // Calculate net value of prizes
+        uint256 valueDistributed;
+        for (uint256 i = 0; i < payoutValues.length; i++) {
+            valueDistributed += payoutValues[i];
+        }
+
+        payoutValues[_prizeValues.length] = _rake; // add the new value at the end
+
+        // Distribute net prizes to players
+        chips.distributeChips(receipients, payoutValues);
+        emit RakeCollected(_gameNumber, _rake);
+
+        currentGameValue[_gameNumber] -= valueDistributed;
+        emit ValueDistributed(_gameNumber, valueDistributed);
+    }
+
+    function _getPlayCost(uint256 _gameNumber) internal view returns(uint256) {
+        return (playCost[_gameNumber] > 0 ? playCost[_gameNumber] : defaultPlayCost);
+    }
+
+    function getPlayCost(uint256 _gameNumber) external view returns(uint256) {
+        return _getPlayCost(_gameNumber);
+    }
+
+    function setPlayCost(uint256 _gameNumber, uint256 _playCost) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
Please add the same roles that the rest of contracts, this should be managed by the MANAGER_ROLE